pkg/genericapiserver/genericapiserver.go

```diff
@@ -547,3 +553,40 @@ func waitForSuccessfulDial(https bool, network, address string, timeout, interva
 	}
 	return err
 }
+
+// getNamedCertificateMap returns a map of strings to *tls.Certificate, suitable for use in
+// tls.Config#NamedCertificates. Returns an error if any of the certs cannot be loaded.
+// Returns nil if len(namedKeyCerts) == 0
+func getNamedCertificateMap(namedKeyCerts []NamedKeyCert) (map[string]*tls.Certificate, error) {
+	if len(namedKeyCerts) == 0 {
+		return nil, nil
+	}
+	namedCerts := map[string]*tls.Certificate{}
+	for _, nkc := range namedKeyCerts {
+		cert, err := tls.LoadX509KeyPair(nkc.CertFile, nkc.KeyFile)
+		if err != nil {
+			return nil, err
+		}
+		if len(nkc.Names) > 0 {
+			for _, name := range nkc.Names {
```

Comment:
explicitly specified names should trump auto-populated names

for example, if I have the following certs:
cert1: CN=bar.com
cert2: CN=foo.com, dnsSAN=bar.com

and I specify these flags:
`--tls-sni-cert-key=cert1.pem,key1.pem,bar.com --tls-sni-cert-key=cert2.pem,key2.pem`

I should end up with
"bar.com": cert1
"foo.com": cert2